ppcs/ppcTODO-more-list-utils.md

```diff
@@ -0,0 +1,125 @@
+# More list processing operators inspired by List::Util
+
+## Preamble
+
+    Author:  Paul Evans <PEVANS>
+    Sponsor: 
+    ID:      TODO
+    Status:  Exploratory
+
+## Abstract
+
+Creates several new list-processing operators, similar to the existing `grep`, inspired by the same-named functions in modules like `List::Util` or `List::Keywords`.
+
+## Motivation
+
+Most code of any appreciable size tends to make use of at least the `any` or `all` functions from `List::Util`. Due to limits of their implementation they are not as efficient to call as core's `grep` operator. The implementations provided by `List::Keywords` are more efficient, on the same level as core's `grep`, though being a non-core module it does not appear to be used anywhere near as much in practice.
+
+## Rationale
+
+## Specification
+
+New named features that, when enabled, activate syntax analogous to the existing `grep` operator, named `any` and `all`:
+
+```
+any { BLOCK } LIST
+
+all { BLOCK } LIST
+```
+
+These operators are similar to `grep` in scalar context, though yield a simple boolean truth value relating to how many input values made the filter block yield true. `any` yields true when its block yields true for at least one of its input values, or false if they all yield false. `all` yields true only if every input value makes the block yield true, or false if at least one yields false.
+
+A consequence of these rules is what happens when given an empty list. A call to `any` with an empty list does not have any input values which made the block return true, so its result is false. Conversely, a call to `all` with an empty list does not have any input values which made the block return false, so its result is true.
```

Comment:
I was at first surprised by this behavior, that all would be true on an empty list; though i did see that that's how `every` works in javascript and `all` in python, and i suppose it makes sense. I think we could have a word on why this design choice makes sense from a user's perspective rather than just the implementation reason.